backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/ConvertDiskCommand.java

```diff
@@ -167,11 +167,21 @@ protected void executeCommand() {
             return;
         }
 
-        long requiredSize =  actionReturnValue.getActionReturnValue();
+        long requiredSize = actionReturnValue.getActionReturnValue();
+
+        CreateVolumeContainerCommandParameters parameters = createVolumeCreationParameters(getDiskImage(), requiredSize);
+        StorageDomain storageDomain = storageDomainDao.get(getLocationInfo().getStorageDomainId());
+
+        // Initial size will not be used for this configuration, meaning the required size will not
+        // be used at all. However, when converting to COW/Preallocated the required size may larger
+        if (parameters.getVolumeType() == VolumeType.Preallocated &&
+                storageDomain.getStorageType().isBlockDomain()) {
```

Comment:
IIRC the initial size is only used for thin-provisioned volumes on block storage - in that case, don't we want to set it also for preallocated volumes on file storage?

Reply:
ack, now checking if the target is preallocated and qcow instead